src/main/java/seedu/exercise/model/ModelManager.java

```diff
@@ -216,8 +227,40 @@ public boolean equals(Object obj) {
         // state check
         ModelManager other = (ModelManager) obj;
         return exerciseBook.equals(other.exerciseBook)
-                && userPrefs.equals(other.userPrefs)
-                && filteredExercises.equals(other.filteredExercises)
-                && filteredRegimes.equals(other.filteredRegimes);
+            && regimeBook.equals(other.regimeBook)
+            && userPrefs.equals(other.userPrefs)
+            && filteredExercises.equals(other.filteredExercises)
+            && filteredRegimes.equals(other.filteredRegimes)
+            && propertyManager.equals(other.propertyManager);
+    }
+
+    //=========== Property Manager Accessors =============================================================
+
+    public PropertyManager getPropertyManager() {
+        return propertyManager;
+    }
+
+    public boolean isPrefixPresent(Prefix prefix) {
+        return propertyManager.isPrefixPresent(prefix);
+    }
+
+    public boolean isFullNamePresent(String fullName) {
+        return propertyManager.isFullNamePresent(fullName);
+    }
+
+    /**
+     * Adds the new prefix into the PropertyManager. PropertyManager will update the property prefixes.
+     */
+    public void addPrefix(Prefix prefix) {
+        propertyManager.addPrefix(prefix);
+        propertyManager.updatePropertyPrefixes();
+    }
+
+    public void addFullName(String fullName) {
+        propertyManager.addFullName(fullName);
+    }
+
+    public void addCustomProperty(CustomProperty customProperty) {
+        propertyManager.addCustomProperty(customProperty);
```

Comment:
Would suggest shifting this entire block for PropertyManagerAccessors to before the ModelManager's `equals()` method.

Reply:
Have made the changes 👍